### DISCUSSION TITLE
Mill 0.10.0-M2: error downloading (404)

### DISCUSSION THREAD

**Discussion by hmf** [com-lihaoyi/mill#1482] [Q&A] [answered]
Hi,

I am trying to use the latest milestone using both the M2 mill script and that script but with the `.mill-version` file.  In the `.mill-version` file I set `0.10.0-M2`. In both cases I get:

```bash
user@node:~/IdeaProjects/commonplace$ ./mill -i files.runMain files.DuplicateContent
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   145  100   145    0     0    423      0 --:--:-- --:--:-- --:--:--   423
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 
```
If I set `0.9.9` in the `.mill-version` file, all works correctly.

Is their anything special we must do to use the milestones?

TIA

**Comment by lefou** [comment, marked as answer]
You forgot to mention what your `./mill` is. A script, the script from mill repository, lefou/millw, anything else?

I assume you're using on of the latter scripts. This issue is already solved for [lefou/millw](https://github.com/lefou/millw) in version 0.3.8 which I recommend to use. A fix for the mill script in the mill repository isn't available yet.

__Edit:__ Oh, the issue is a slighly more complicatad download URL due to the newly introduce milestone part in the version tag.

**Reply by hmf** [reply]
I am using the script in that is [here](https://github.com/com-lihaoyi/mill/releases/download/0.10.0-M2/0.10.0-M2) (mill repository). I use this in Linux. 

Note that the script works if I set it to the `0.9.9` version. Its the milestone that fails.

EDIT: strange, I was under the impression the `millw` was specifically designed for windows.

**Reply by lefou** [reply]
No, it's more a generic launcher for mill, supporting almost any (old) version (too). The 'mill**w**' stands for mill-**wrapper**, as usually used in other tools: gradlew, mvnw, ...

**Reply by hmf** [reply]
Ok. Thanks for the info. As soon as I can I will try with `millw` and report back.

**Reply by hmf** [reply]
@lefou millw worked. Thank you.
